openedx/openedx-platform
# [SE-3329] Expose banner image url in course API

This PR extends the course overview and detail APIs to include (both relative and absolute) banner image url, which allows content publishers to specify a banner image and then reuse that from a custom UI/integration.

**Dependencies**: None

**Screenshots**:

![Screenshot_2020-11-12 Course Detail – Django REST framework](https://user-images.githubusercontent.com/19173947/99000523-b382f600-2539-11eb-84ea-61526843c73a.png)
![Screenshot_2020-11-12 Course List – Django REST framework](https://user-images.githubusercontent.com/19173947/99000525-b4b42300-2539-11eb-8379-6d219470e72b.png)


**Sandbox URL**: https://pr25045.sandbox.opencraft.hosting/

**Merge deadline**: Preferably 29th of September

**Testing instructions**:

1. Open edX Studio from **Content -> Files & Uploads**, then copy the file name
2. Paste the file name into `Course Banner Image` field into **Settings -> Advanced Settings** page
3. Open `<URL>/api/courses/v1/courses/` to check all the courses, where URL is the sandbox LMS URL
4. Open `<URL>/api/courses/v1/courses/<course_id>` to see all the courses, where URL is the sandbox LMS URL and course_id is the course id

**Author notes and concerns**:

1. The `_AbsolutMediaSerializer` is using a private attribute from `AbsoluteURLField`, but I found no other way to avoid code duplication and have the same absolute url conversion functionality, since `AbsoluteURLField` requires the request object
   Currently looking for ways to fix it.

**Reviewers**

**Settings**
```yaml
EDXAPP_FEATURES:
  ENABLE_COMBINED_LOGIN_REGISTRATION: true
```
